MushroomObserver/mushroom-observer
# Normalize Sequences Controller

This PR changes `SequenceController`, and associated files, routes, actions, and views to the Rails defaults, including
- pluralizes "Sequence";
- drops "sequence" prefix from file and method names;
- `index` (rather than `list`) as the only index action,
combining legacy indexing actions into "flavors" of normalized `index`;
- passes Observation id to `new` and `create` as a query param, rather than a route param, 
i.e. ~~`/sequence/create_sequence/123`~~ => `sequences/new?obs_id=123`,
changing routes accordingly;
- uses `form_with` (not `form_for`), extracting form builder calls to the form file;
- changes controller test to exclusively "request" tests, i.e., one request per method;
- `destroy` UI is a button, rather than a link;
- a variety of other improvements and fixes.

### Suggested Manual Test
Manual test
- Goto any Observation
- Try to create a Sequence with a Locus (any text will do) but without other attributes.
Expected result: Flash error, returned to new Sequence form
- Add the bases (any combination of ACGT, spaces, returns).
Expected result: Sequence created, returned to Observation
- Edit the Sequence (whatever you like).
Expected result: Update successful, returned to Sequence
- Destroy Sequence (in Dev Mode this must be confirmed 2x).
Expected result: Sequence destroyed, returned to Observation
- http://localhost:3000/sequences
Expected result: complete Sequence index
